antarest/study/storage/variantstudy/model/command/update_binding_constraint.py

```diff
@@ -1,22 +1,108 @@
+    BindingConstraintOperator,
+)
 from antarest.study.storage.rawstudy.model.filesystem.config.model import FileStudyTreeConfig
 from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+from antarest.study.storage.rawstudy.model.filesystem.lazy_node import LazyNode
 from antarest.study.storage.variantstudy.model.command.common import CommandName, CommandOutput
 from antarest.study.storage.variantstudy.model.command.create_binding_constraint import (
     DEFAULT_GROUP,
-    TERM_MATRICES,
     AbstractBindingConstraintCommand,
+    TermMatrices,
     create_binding_constraint_config,
 )
 from antarest.study.storage.variantstudy.model.command.icommand import MATCH_SIGNATURE_SEPARATOR, ICommand
 from antarest.study.storage.variantstudy.model.model import CommandDTO
 
-MatrixType = List[List[MatrixData]]
+MatrixType = t.List[t.List[MatrixData]]
+
+ALIAS_OPERATOR_MAP = {
+    BindingConstraintOperator.EQUAL: "eq",
+    BindingConstraintOperator.LESS: "lt",
+    BindingConstraintOperator.GREATER: "gt",
+}
+
+HANDLED_OPERATORS = [
+    BindingConstraintOperator.EQUAL,
+    BindingConstraintOperator.LESS,
+    BindingConstraintOperator.GREATER,
+    BindingConstraintOperator.BOTH,
+]
+
+
+def _update_matrices_names(
+    file_study: FileStudy,
+    binding_constraint_id: str,
+    existing_operator: BindingConstraintOperator,
+    new_operator: BindingConstraintOperator,
+) -> None:
+    """
+    Update the matrix file name according to the new operator.
+
+    Args:
+        file_study: the file study
+        binding_constraint_id: the binding constraint ID
+        existing_operator: the existing operator
+        new_operator: the new operator
+
+    Raises:
+        NotImplementedError: if the case is not handled
+    """
+
+    parent_folder_node = file_study.tree.get_node(["input", "bindingconstraints"])
+    matrix_lt = parent_folder_node.get_node([f"{binding_constraint_id}_lt"])
+    assert isinstance(matrix_lt, LazyNode), f"Node type not handled yet: LazyNode expected, got {type(matrix_lt)}"
+    matrix_eq = parent_folder_node.get_node([f"{binding_constraint_id}_eq"])
+    assert isinstance(matrix_eq, LazyNode), f"Node type not handled yet: LazyNode expected, got {type(matrix_eq)}"
+    matrix_gt = parent_folder_node.get_node([f"{binding_constraint_id}_gt"])
+    assert isinstance(matrix_gt, LazyNode), f"Node type not handled yet: LazyNode expected, got {type(matrix_gt)}"
+
+    # TODO: due to legacy matrices generation, we need to check if the new matrix file already exists
+    #  and if it does, we need to first remove it before renaming the existing matrix file
+
+    if (existing_operator not in HANDLED_OPERATORS) or (new_operator not in HANDLED_OPERATORS):
+        raise NotImplementedError(
+            f"Case not handled yet: existing_operator={existing_operator}, new_operator={new_operator}"
+        )
+    elif existing_operator == new_operator:
+        return  # nothing to do
```

Comment:
Shouldn't we remove matrices that are not concerned by the operator in such a case due to legacy matrices generation ?

Reply:
In previous discussion, I explained that I prefer to delete only when you need to not because you want to.